Loggie/Classes/LogList/LogListTableViewController.swift

```diff
@@ -50,29 +59,73 @@ public class LogListTableViewController: UITableViewController {
         tableView.register(cellNib, forCellReuseIdentifier: LogListTableViewController.cellReuseIdentifier)
     }
 
-    // MARK: - Table view data source
+    @IBAction func clearAction() {
+        logsDataSourceDelegate.clearLogs()
+        logs = []
+        searchResult = []
+    }
+}
+
+// MARK: - Table view data source
+
+extension LogListTableViewController {
 
     override public func numberOfSections(in tableView: UITableView) -> Int {
         return 1
     }
 
     override public func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return logs.count
+        return dataSource.count
     }
 
     override public func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: LogListTableViewController.cellReuseIdentifier, for: indexPath) as! LogListTableViewCell
-        cell.configure(with: logs[indexPath.row])
+        let cell = tableView.dequeueReusableCell(
+            withIdentifier: LogListTableViewController.cellReuseIdentifier,
+            for: indexPath
+        ) as! LogListTableViewCell
+
+        cell.configure(with: dataSource[indexPath.row])
         return cell
     }
+}
+
+// MARK: - Table view delegate
 
-    // MARK: - Table view delegate
+extension LogListTableViewController {
 
     override public func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        showLogDetails(with: logs[indexPath.row])
+        showLogDetails(with: dataSource[indexPath.row])
+    }
+
+    public override func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        if searchBar.isFirstResponder {
+            searchBar.resignFirstResponder()
+        }
     }
+}
+// MARK: - Search bar delegate
+
+extension LogListTableViewController: UISearchBarDelegate {
+
+    public func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+        isSearching = searchText.isEmpty == false
+        guard isSearching else {
+            searchResult = []
+            tableView.reloadData()
+            return
+        }
+        searchResult = logs.filter({ $0.searchableText.range(of: searchText, options: .caseInsensitive) != nil })
```

Comment:
Also, would suggest avoid creating "dummy" string just for search purposes, could be solved something like this:

```swift
var searchKeywords: [String] {
    return [
        request.url?.absoluteString,
        response.flatMap { String($0.statusCode) }
    ].compactMap { $0 }
}
...
let shouldInclude: (Log) -> Bool = {
    $0.searchKeywords.contains(where: {
        $0.range(of: searchText, options: .caseInsensitive) != nil
    })
}
searchResult = logs.filter(shouldInclude)
```